src/outfit.ts

```diff
@@ -0,0 +1,125 @@
+import { OutfitSlot, OutfitSpec } from "grimoire-kolmafia";
+import {
+  canEquip,
+  canInteract,
+  Familiar,
+  inebrietyLimit,
+  Item,
+  itemAmount,
+  myInebriety,
+} from "kolmafia";
+import { $familiar, $familiars, $item, $items, get, have, sumNumbers } from "libram";
+
+import { ChronerQuest } from "./engine";
+import { freeFightFamiliar, MenuOptions } from "./familiar";
+import { garboAverageValue, garboValue } from "./garboValue";
+import { realmAvailable } from "./lib";
+
+export function ifHave(slot: OutfitSlot, item: Item, condition?: () => boolean): OutfitSpec {
+  return have(item) && canEquip(item) && (condition?.() ?? true)
+    ? Object.fromEntries([[slot, item]])
+    : {};
+}
+
+const chooseFamiliar = (options: MenuOptions = {}): Familiar =>
+  canInteract() && myInebriety() <= inebrietyLimit()
+    ? $familiars`Reagnimated Gnome, Temporal Riftlet`.find((f) => have(f)) ??
+      freeFightFamiliar(options)
+    : freeFightFamiliar(options);
+
+export function chooseQuestOutfit(quest: ChronerQuest, outfit: OutfitSpec = {}): OutfitSpec {
```

Comment:
can you make this `...outfitSpecs: OutfitSpec[]` instead, and merge it in this function? or is that too insane

Reply:
yeah, I'll `reduce`